apollo-federation/src/schema/validators/interface_object.rs

```diff
@@ -0,0 +1,167 @@
+use crate::error::FederationError;
+use crate::error::MultipleFederationErrors;
+use crate::error::SingleFederationError;
+use crate::schema::ValidFederationSchema;
+use crate::schema::field_set::parse_field_set;
+use crate::schema::position::InterfaceTypeDefinitionPosition;
+use crate::schema::subgraph_metadata::SubgraphMetadata;
+use crate::utils::human_readable::HumanReadableListOptions;
+use crate::utils::human_readable::HumanReadableListPrefix;
+use crate::utils::human_readable::human_readable_list;
+
+pub(crate) fn validate_interface_object_directives(
+    schema: &ValidFederationSchema,
+    metadata: &SubgraphMetadata,
+    errors: &mut MultipleFederationErrors,
+) -> Result<(), FederationError> {
+    validate_keys_on_interfaces_are_also_on_all_implementations(schema, metadata, errors)?;
```

Comment:
Strictly speaking, `validate_keys_on_interfaces_are_also_on_all_implementations` is not about interface object. 😓  Not a big deal, though.

Reply:
Interface `@key`s (a.k.a. [entity interfaces](https://www.apollographql.com/docs/graphos/schema-design/federated-schemas/entities/interfaces)) were a feature introduced specifically to facilitate interface objects, which is why they're grouped together (changes to one will need to be considered in tandem with the other).